src/traits.rs

```diff
@@ -67,6 +67,40 @@ impl Enlargeable for u32 {
     type Larger = u64;
 }
 
+/// Linear interpolation without involving floating numbers.
+pub trait Lerp: Sized + Bounded + NumCast {
```

Comment:
I just want to note that I'm not 100% sure about the design here but it seems like a minor detail right now without enough room for adjustment so this shouldn't hold up the PR on its own. My minor pain points are:

* Many of the bounds exist only for the purpose of the default impl. In particular the `SignedLarger` type isn't actually supposed to be used by the caller which is odd for an associated type.
* The requirements on `SignedLarger` are more intricate than `Enlargeable`. A comment wouldn't hurt.

Reply:
But I do see the design similar to `Enlargeable` and it is not a critical center piece of the crate so even if we _want_ to change it in the future we can deprecate and replace it without too much of a fuzz. So nothing you have to act on :)

Reply:
Renamed `SignedLarger` to `Ratio` to reflect floating point usage.